tests/JavaTest.java

```diff
@@ -97,7 +97,7 @@ public static void main(String[] args) {
 
         TestSizedInputStream();
 
-        TestVectorOfUnions();
+//        TestVectorOfUnions();
```

Comment:
wait, why is this commented out?

Reply:
There is that issue I opened last week about non Table in Union. This test doesn't fail but it will when we add ObjectAPI as there is no matching methods for the non Tables.
It should be fixed than added back.

Reply:
We should fix that here.. we can't just disable existing tests.

Reply:
OK. So here is my solution.
For now, ObjectAPI won't support Unions of non-tables just like the other API doesn't.
The moment issue #6561 will be fixed I will add that support back based on the new API that will be defined in this issue.
I will re-enable these tests.

Reply:
> General note: we're going to be releasing FlatBuffers 2.0 very soon, and since there are still things to be figured out in this PR, would be good to land it after 2.0, so it will be easier to fix issues if any arise.

In my opinion, it contradicts the idea of semantic versioning. The ObjectAPI can be implemented (and already is) for the the current version without breaking the existing API, so why denying users of that version from its benefits.

I'm not that aware of the changes version 2 will make, and if there are some which will break this code, I'm not sure I will be able to invest time in it.
This feature took me very long time to deliver, not to mention how many times it was requested and wasn't implemented at the end.